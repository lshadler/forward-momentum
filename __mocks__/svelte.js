const svelte = jest.requireActual('svelte');

module.exports = {
  ...svelte,
  getContext: jest.fn(() => ({
    open: jest.fn(),
  })),
};
