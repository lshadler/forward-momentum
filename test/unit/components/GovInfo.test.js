import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/svelte';

import Comp from '../../../src/components/GovInfo.svelte';

describe('GovInfo', () => {
  test('matches snapshot', () => {
    const { container } = render(Comp, { term: { contact_form: 'test' }, id: { bioguide: 'test', fec: [] } });

    expect(container.innerHTML).toMatchSnapshot();
  });
});
