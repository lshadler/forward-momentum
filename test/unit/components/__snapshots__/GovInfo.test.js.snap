// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovInfo rep - matches snapshot 1`] = `"<div>   </div>"`;

exports[`GovInfo sen - matches snapshot 1`] = `"<div><a class=\\"blue fa fa-id-card svelte-1b6o2g4\\" target=\\"_blank\\" rel=\\"noopener noreferer\\" href=\\"test\\">&nbsp;<span class=\\"svelte-1b6o2g4\\">Online contact form</span></a> <a class=\\"red fa fa-check-square-o svelte-1b6o2g4\\" target=\\"_blank\\" rel=\\"noopener noreferer\\" href=\\"https://justfacts.votesmart.org/candidate/key-votes/test\\">&nbsp;<span class=\\"svelte-1b6o2g4\\">View on VoteSmart</span></a> <a class=\\"blue fa fa-usd svelte-1b6o2g4\\" target=\\"_blank\\" rel=\\"noopener noreferer\\" href=\\"https://www.fec.gov/data/candidates?q=test\\">&nbsp;<span class=\\"svelte-1b6o2g4\\">View campaign info</span></a> <a class=\\"red fa fa-bar-chart-o svelte-1b6o2g4\\" target=\\"_blank\\" rel=\\"noopener noreferer\\" href=\\"https://www.govtrack.us/congress/members/test\\">&nbsp;<span class=\\"svelte-1b6o2g4\\">GovTrack</span></a></div>"`;
